### DISCUSSION TITLE
How to get the collection type and id associated to a item URL ?

### DISCUSSION THREAD

**Discussion by ghost** [VirtusLab/strapi-plugin-navigation#178] [Q&A] [answered]
Hi,

From the api endpoint given by the navitation plugin, I can't build my link in front with the corresponding URL.

What is the best practice to, from the URL, get the associated collection type and id ?

For now, I'm just loop into the full tree to find it, but I gess there is a better way.

**Comment by ltsNotMike** [comment, marked as answer]
Hello,

Our plugin has a `render` endpoint that returns a tree or a list of all navigation items. There you can find a `related` field with both id and collection type of the content types associated with your navigation items. 

You can also apply the `path` parameter to the `render` endpoint to render only a subtree of navigation items. You can read all about it [here](https://github.com/VirtusLab-Open-Source/strapi-plugin-navigation#%EF%B8%8F-public-api-specification). I hope this will help you but if you have further questions feel free to ask them and I will try my best to answer them. 🙂

**Reply by ghost** [reply]
Hi, 

I read the doc but I didn't anterstood that point!

Thank you !
